hazelcast/src/main/java/com/hazelcast/internal/networking/ChannelOutboundPipeline.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.networking;
+
+/**
+ * The outbound pipeline of a {@link Channel}. So all data that gets
+ * written to the network, goes through the outbound pipeline.
+ */
+public interface ChannelOutboundPipeline {
+
+
+      /**
+     * Adds the handlers at the end of the pipeline
+     *
+     * No verification is done if the handler is already added and a handler
+     * should only be added once.
+     *
+     * This method should only be made on the thread 'owning' the handler.
+     *
+     * @param handlers the handlers to add.
+     * @return this
+     */
+    ChannelOutboundPipeline addLast(ChannelOutboundHandler... handlers);
+
+      /**
+     * Replaces the old ChannelOutboundHandler by the new ones.
+     *
+     * No verification is done if any of the handlers is already added and a
+     * handler should only be added once.
+     *
+     * This method should only be made on the thread 'owning' the handler.
+     *
+     * If the oldHandler doesn't exist, the call is ignored
```

Comment:
Shouldn't call fail when oldHandler does not exist? Otherwise, caller will not aware if replace(..) succeeded or not.

Reply:
Fixed

Reply:
Same as `ChannelInboundPipeline`, javadoc still says "_the call is ignored_".